crates/oxc_linter/src/rules/import/extensions.rs

```diff
@@ -179,220 +326,407 @@ declare_oxc_lint!(
 impl Rule for Extensions {
     fn from_configuration(value: serde_json::Value) -> Self {
         if let Some(first_arg) = value.get(0).and_then(Value::as_str) {
-            let default = FileExtensionConfig::from(first_arg);
+            let default = ExtensionRule::from_str(first_arg);
 
             if let Some(val) = value.get(1) {
                 let root = val.get("pattern").unwrap_or(val);
 
-                let config = build_config(root, Some(&default));
+                let config = build_config(root, default);
 
                 Self(Box::new(config))
             } else {
-                let config = build_config(&value, Some(&default));
+                let config = build_config(&value, default);
 
                 Self(Box::new(config))
             }
+        } else if let Some(first_obj) = value.get(0) {
+            // First element is not a string, but is present (e.g., [{ "json": "always" }])
+            let config = build_config(first_obj, None);
+            Self(Box::new(config))
         } else {
             let config = build_config(&value, None);
             Self(Box::new(config))
         }
     }
 
     fn run_once(&self, ctx: &LintContext) {
-        let module_record = ctx.module_record();
-
-        let config = self.0.clone();
-
+        // Process require() calls
         for node in ctx.nodes().iter() {
```

Comment:
Could we move this to an additional `fn run` implementation instead? Rules can implement both `run` and `run_once`, and that would take the explicit node iteration out of this. That's helpful to do because we can optimize how often we call `run`.